File: application/src/main/vue/src/domain/value_objects/rank.ts
export enum Rank {
  NONE = 0,
  FIRST = 1,
  SECOND = 2,
  THIRD = 3,
  FOURTH = 4,
}

export function rankOf(ordinal: number): Rank {
  if (ordinal === 1) {
    return Rank.FIRST;
  } else if (ordinal === 2) {
    return Rank.SECOND;
  } else if (ordinal === 3) {
    return Rank.THIRD;
  } else if (ordinal === 4) {
    return Rank.FOURTH;
  } else {
    return Rank.NONE;
  }
}

File: domain/src/main/java/ch/resrc/tichu/domain/value_objects/Rank.java
package ch.resrc.tichu.domain.value_objects;

import java.util.*;

public enum Rank {
    NONE(0),
    FIRST(1),
    SECOND(2),
    THIRD(3),
    FOURTH(4);

    private final int value;

    Rank(int value) {
        this.value = value;
    }

    public static Rank of(int value) {
        return Arrays.stream(Rank.values())
                .filter(rank -> rank.value == value)
                .findFirst()
                .orElse(Rank.NONE);
    }

    public int value() {
        return value;
    }
}
